src/scriptInterfaceMagic.h

```diff
@@ -227,21 +227,58 @@ template<typename T> struct convert<std::vector<T> >
             T var;
             convert<T>::param(L, idx, var);
             v.push_back(var);
-        }
-    }
+        }
+    }
+
+    static int returnType(lua_State* L, std::vector<T> vector)
+    {
+        lua_newtable(L);
+        int n = 1;
+        for(auto val : vector)
+        {
+            lua_pushnumber(L, n);
+            if (convert<T>::returnType(L, val))
```

Comment:
returnType can push more then 2 entries on the stack, with sf::Vector2 for example. Which will break this code.

Reply:
Valid concern. 

I added checks that just use the first return value and issue a warning when not exactly one value is put on the stack. This should make sure that the stack stays clean.

I thought about implementing support for this cases, but its rather complex and should be thoroughly tested. You would need to add a new table to the stack, then move it below all returnValues, then reverse the order of all values on the stack, because you can only commit values from the top of the stack, and submit them to the table. It's quite some hazzle that is not needed at the moment. I would implement and test it when it is first needed.